robottelo/hosts.py

```diff
@@ -1752,14 +1752,26 @@ def __init__(self, hostname=None, **kwargs):
         self._api = type('api', (), {'_configured': False})
         self._cli = type('cli', (), {'_configured': False})
 
+    def _swap_nailgun(self, new_version):
+        """Install a different version of nailgun from GitHub and invalidate the module cache."""
+        import sys
+
+        from pip._internal import main as pip_main
+
+        pip_main(['uninstall', '-y', 'nailgun'])
+        pip_main(['install', f'https://github.com/SatelliteQE/nailgun/archive/{new_version}.zip'])
+        self._api = type('api', (), {'_configured': False})
+        to_clear = [k for k in sys.modules.keys() if 'nailgun' in k]
+        [sys.modules.pop(k) for k in to_clear]
+
     @property
     def api(self):
         """Import all nailgun entities and wrap them under self.api"""
         if not self._api:
             self._api = type('api', (), {'_configured': False})
         if self._api._configured:
             return self._api
-
+        from nailgun import entities as _entities  # use a private import
```

Comment:
Not sure how this would help with nailgun switching, could you help us to understand ?

Reply:
`entities` are already imported on line 23 of this file, so we need a new alias to use the new import.

Reply:
But wouldnt simply removing the old nailgun and installing a new nailgun point the entities in a new nailgun module ? Do we really need to differ here ?

Reply:
no because `entities` is a live object already loaded into memory in this namespace so it will persist. You can test this out locally.
```python
import sys
from nailgun import entities
print("===== Before =====")
print(id(entities), id(entities.ActivationKey))
to_clear = [k for k in sys.modules.keys() if 'nailgun' in k]
[sys.modules.pop(k) for k in to_clear]
print("===== After =====")
print(id(entities), id(entities.ActivationKey))
```